c++20 related test filure on ix86 (32 bit intel)

### Check duplicate issues.

- [x] Checked for duplicates

### Description

Fedora rawhide recently upgraded to gcc 16, which uses c++20 as default.
There are 2 new test failures on ix86 (32 bit intel)
```
 219/1554 Test  #169: gtest-hist-hist-testTH3 .......................................................................***Failed    0.26 sec
Running main() from /builddir/build/BUILD/gtest-1.17.0-build/googletest-1.17.0/googletest/src/gtest_main.cc
Note: Google Test filter = -RNTuple.StdAtomic:RCsvDS.Remote:RFile.RemoteRead:RNTuple.OpenHTTP:RRawFile.Remote:RSqliteDS.Davix:TChainParsing.DoubleSlash:TChainParsing.RemoteGlob:TFile.ReadWithoutGlobalRegistrationNet:TFile.ReadWithoutGlobalRegistrationWeb:TTreeRegressions.PrintClustersRounding
[==========] Running 3 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 1 test from TH3L
[ RUN      ] TH3L.SetBinContent
[       OK ] TH3L.SetBinContent (0 ms)
[----------] 1 test from TH3L (0 ms total)
[----------] 2 tests from TH3D
[ RUN      ] TH3D.FillThreadSafe
/usr/include/c++/16/atomic:1772: std::atomic_ref<_Tp>::atomic_ref(_Tp&) [with _Tp = double]: Assertion '((unsigned int)this->_M_ptr % this->required_alignment) == 0' failed.
/usr/include/c++/16/atomic:1772: std::atomic_ref<_Tp>::atomic_ref(_Tp&) [with _Tp = double]: Assertion '((unsigned int)this->_M_ptr % this->required_alignment) == 0' failed.
CMake Error at /builddir/build/BUILD/root-6.38.00-build/root-6.38.00/cmake/modules/RootTestDriver.cmake:232 (message):
  error code: Subprocess aborted
```
```
589/1554 Test  #549: gtest-tree-dataframe-dataframe-helpers ........................................................***Failed    4.83 sec
Running main() from /builddir/build/BUILD/gtest-1.17.0-build/googletest-1.17.0/googletest/src/gtest_main.cc
Note: Google Test filter = -RNTuple.StdAtomic:RCsvDS.Remote:RFile.RemoteRead:RNTuple.OpenHTTP:RRawFile.Remote:RSqliteDS.Davix:TChainParsing.DoubleSlash:TChainParsing.RemoteGlob:TFile.ReadWithoutGlobalRegistrationNet:TFile.ReadWithoutGlobalRegistrationWeb:TTreeRegressions.PrintClustersRounding
[==========] Running 31 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 25 tests from RDFHelpers
[ -- successful tests removved -- ]
[ RUN      ] RDFHelpers.Histo3DThreadSafe
/usr/include/c++/16/atomic:1772: std::atomic_ref<_Tp>::atomic_ref(_Tp&) [with _Tp = double]: Assertion '((unsigned int)this->_M_ptr % this->required_alignment) == 0' failed.
/usr/include/c++/16/atomic:1772: std::atomic_ref<_Tp>::atomic_ref(_Tp&) [with _Tp = double]: Assertion '((unsigned int)this->_M_CMake Error at /builddir/build/BUILD/root-6.38.00-build/root-6.38.00/cmake/modules/RootTestDriver.cmake:232 (message):
  error code: Subprocess aborted
```
Both tests are related to TH3D class and thread safe filling, and fail with alignment problems for atomic_ref<doubke>. which is a c++20 feature.

### Reproducer

Build root 6.38.00 for ix86 with gcc 16. Run tests.

### ROOT version

6.38.00

### Installation method

built from source (rpm package build)

### Operating system

Fedora Rawhide (Fedora 44) for 32 bit intel

### Additional context

Compiler installed from Fedora rawhide rpm: gcc-c++-16.0.0-0.4.fc44.i686

--- a/hist/hist/inc/TH3.h
+++ b/hist/hist/inc/TH3.h
@@ -28,6 +28,11 @@
 class TH2D;
 class TProfile2D;
 
+#if defined R__B64 && defined __cpp_lib_atomic_ref
+// ROOT-20834 Atomic_ref on 32 bit can fail because of alignment problems
+#define TH3D_FILL_THREADSAFE
+#endif
+
 namespace ROOT::Internal {
 /// Entrypoint for thread-safe filling from RDataFrame.
 template <typename T, typename... Args>
@@ -455,7 +460,7 @@ protected:
            Double_t RetrieveBinContent(Int_t bin) const override { return fArray[bin]; }
            void     UpdateBinContent(Int_t bin, Double_t content) override { fArray[bin] = content; }
 private:
-#ifdef __cpp_lib_atomic_ref
+#ifdef TH3D_FILL_THREADSAFE
            void FillThreadSafe(Double_t x, Double_t y, Double_t z, Double_t w = 1.);
            template <typename T, typename... Args>
            friend auto ROOT::Internal::FillThreadSafe(T &histo, Args... args)

--- a/hist/hist/src/TH3.cxx
+++ b/hist/hist/src/TH3.cxx
@@ -438,7 +438,7 @@ Int_t TH3::Fill(Double_t x, Double_t y, Double_t z, Double_t w)
    return bin;
 }
 
-#ifdef __cpp_lib_atomic_ref
+#ifdef TH3D_FILL_THREADSAFE
 ////////////////////////////////////////////////////////////////////////////////
 /// Atomically increment cell defined by x,y,z by a weight w.
 ///

--- a/hist/hist/test/test_TH3.cxx
+++ b/hist/hist/test/test_TH3.cxx
@@ -14,7 +14,7 @@ TEST(TH3L, SetBinContent)
    EXPECT_EQ(h.GetBinContent(1, 1, 1), Large);
 }
 
-#ifdef __cpp_lib_atomic_ref
+#ifdef TH3D_FILL_THREADSAFE
 
 TEST(TH3D, FillThreadSafe)
 {

